### ISSUE TITLE
Support for Svelte 5

### ISSUE THREAD

**Issue by dshook** [extension-js/extension.js#220]
Hi there, thanks for the project!

The big new version of Svelte came out a few weeks ago and was wondering if there's plans to upgrade to support it.

Looks like [svelte-preprocess](https://www.npmjs.com/package/svelte-preprocess) will need to be upgraded.  

I was able to get it working by adding the resolution in package.json
```
  "resolutions": {
    "svelte-preprocess": "6.0.3"
  }
```
And updating the `scripts.ts` in the svelte example to use the new mount syntax.

**Comment by cezaraugusto** [comment]
hey @dshook, totally want this.

cc @akadotsh - interested to take this one?

**Comment by cezaraugusto** [comment]
@dshook could you share a sample web app using Svelte 5 so I can take a look? not a Svelte dev myself

**Comment by dshook** [comment]
You should be able to find plenty of other examples online, but for extension-js you can just mount one in a content script:

```
import { mount } from "svelte";
import Test from "../Test.svelte";

const container = document.querySelector('.wherever');

const app = mount(Test, {
  target: container,
});

```

Where Test.svelte is something like

```
<script>
	let count = $state(0);
</script>

<button onclick={() => count++}>
	clicks: {count}
</button>
```

**Comment by dandv** [comment]
> could you share a sample web app using Svelte 5

https://github.com/NekitCorp/chrome-extension-svelte-typescript-boilerplate is  good Svelte boilerplate, just not cross-browser. The emitted `background` prop in `manifest.json` has a `type: "module"` for instance (even though the generated service_worker is vanilla JS), which Firefox complains about.

**Comment by cezaraugusto** [comment]
working on it

**Comment by cezaraugusto** [comment]
fixed by #236. cc @dshook @dandv please update `extension@2.0.0-beta.8` and run the [Svelte template](https://github.com/extension-js/extension.js/tree/main/examples/new-svelte). please re-open this issue if for any reason it doesn't work on your end. thanks!

**Comment by dshook** [comment]
@cezaraugusto Hi there, sorry for the delay, wasn't able to test it out till today and I'm not able to reopen this issue since you closed it.  Looks like the update works well for new tabs, but I'm having an issue when using Svelte 5 in a content script where everything works the first time, but then after changing anything the hot reload gets into an infinite loop reloading the page.

Here's a repo based off the svelte template: https://github.com/dshook/extension-svelte-test then all you have to do is:
1. go to google
2. See the component works in the top right
3. Make any change to the results component
4. See the page reload over and over again
